src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/FileLayoutOptimizer.cs

```diff
@@ -153,6 +154,10 @@ int ComputeHotColdRegion(MethodWithGCInfo method)
                     }
                     break;
 
+                case MethodLayoutAlgorithm.InstrumentedHotCold:
```

Comment:
It is a fine baby step to add this, but this is in need of redesign. We want to end up with a blended algorithm that does combination of:
- It is enabled by default
- Uses profile data to place methods hit in the profile together (this PR)
- Places hot methods (e.g. methods executed more than a few times in the profile) together
- Sorts methods based on connectedness (ie places methods that call each other close to each other)

Reply:
I think the existing Pettis-Hansen algorithm would do that. The problem is we only know methods that call each other from sampling. Static analysis can only make guesses about the callgraph and in case of virtual methods it's going to be a wild guess. The first party customer we're working with on this should be able to provide good sampling data that we can then try Pettis-Hansen with.

Reply:
> Static analysis can only make guesses about the callgraph and in case of virtual methods it's going to be a wild guess.

Taking advantage of these guesses in the image layout provides measurable improvement based on our past experience with fragile ngen. Fragile ngen had a layout pass driven by a (static) callgraph https://github.com/dotnet/coreclr/blob/release/1.0.0/src/zap/zapimage.cpp#L3641-L3658 .